hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/volumes/volumes.tsx

```diff
@@ -0,0 +1,422 @@
+import {AxiosGetHelper} from "../../utils/axiosRequestHelper";
+
+interface IVolumeResponse {
+  volume: string;
+  owner: string;
+  admin: string;
+  creationTime: number;
+  modificationTime: number;
+  quotaInBytes: number;
+  quotaInNamespace: number;
+  usedNamespace: number;
+  acls: IAcl[];
+}
+
+type VolumnTableColumn = ColumnProps<any> & any;
+
+interface IVolumesResponse {
+  totalCount: number;
+  volumes: IVolumeResponse[];
+}
+
+interface IVolumesState {
+  loading: boolean;
+  dataSource: IVolume[];
+  totalCount: number;
+  lastUpdated: number;
+  selectedColumns: IOption[];
+  columnOptions: IOption[];
+  currentRow?: IVolume;
+  showPanel: boolean;
+  selectedLimit: IOption;
+}
+
+const COLUMNS: VolumnTableColumn[] = [
+  {
+    title: 'Volume',
+    dataIndex: 'volume',
+    key: 'volume',
+    isVisible: true,
+    isSearchable: true,
+    sorter: (a: IVolume, b: IVolume) => a.volume.localeCompare(b.volume),
+    defaultSortOrder: 'ascend' as const,
+    fixed: 'left'
+  },
+  {
+    title: 'Owner',
+    dataIndex: 'owner',
+    key: 'owner',
+    isVisible: true,
+    isSearchable: true,
+    sorter: (a: IVolume, b: IVolume) => a.owner.localeCompare(b.owner)
+  },
+  {
+    title: 'Admin',
+    dataIndex: 'admin',
+    key: 'admin',
+    isVisible: true,
+    isSearchable: true,
+    sorter: (a: IVolume, b: IVolume) => a.admin.localeCompare(b.admin)
+  },
+  {
+    title: 'Creation Time',
+    dataIndex: 'creationTime',
+    key: 'creationTime',
+    isVisible: true,
+    sorter: (a: IVolume, b: IVolume) => a.creationTime - b.creationTime,
+    render: (creationTime: number) => {
+      return creationTime > 0 ? moment(creationTime).format('ll LTS') : 'NA';
+    }
+  },
+  {
+    title: 'Modification Time',
+    dataIndex: 'modificationTime',
+    key: 'modificationTime',
+    isVisible: true,
+    sorter: (a: IVolume, b: IVolume) => a.modificationTime - b.modificationTime,
+    render: (modificationTime: number) => {
+      return modificationTime  > 0 ? moment(modificationTime).format('ll LTS') : 'NA';
+    }
+  },
+  {
+    title: 'Quota (Size)',
+    dataIndex: 'quotaInBytes',
+    key: 'quotaInBytes',
+    isVisible: true,
+    render: (quotaInBytes: number) => {
+      return quotaInBytes && quotaInBytes !== -1 ? byteToSize(quotaInBytes, 3) : 'NA';
+    }
+  },
+  {
+    title: 'Namespace Capacity',
+    key: 'namespaceCapacity',
+    isVisible: true,
+    sorter: (a: IVolume, b: IVolume) => a.usedNamespace - b.usedNamespace,
+    render: (text: string, record: IBucket) => (
+      <QuotaBar
+        quota={record.quotaInNamespace}
+        used={record.usedNamespace}
+        quotaType='namespace'
+      />
+    )
+  },
+  {
+    title: 'Buckets',
+    key: 'listBuckets',
+    isVisible: true,
+    render: (_: any, record: IVolume) => {
+      const searchParams = new URLSearchParams();
+      searchParams.append('volume', record.volume);
+
+      return (
+        <Link to={`/Buckets?${searchParams.toString()}`}>
```

Comment:
Is nested `<a>` tag required?
I would suggest doing it as:
```suggestion
        <Link key="listBuckets" to={`/Buckets?${searchParams.toString()}`}>
            Show buckets
        </Link>
```

Reply:
Thanks for the review. Yes, you are right. Updated as suggested.